common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaNamingStrategy.java

```diff
@@ -20,9 +20,17 @@
  * An implementation of this interface can be provided and passed via property.
  */
 public interface AWSSchemaNamingStrategy {
-    default String getSchemaName(String transportName, Object data) {
+    default String getSchemaName(String transportName,
+                                 Object data) {
         return getSchemaName(transportName);
     }
+
+    default String getSchemaName(String transportName,
+                                 Object data,
+                                 boolean isKey) {
+        return isKey ? getSchemaName(transportName) : getSchemaName(transportName, data);
```

Comment:
So, by default `data` is ignored? I assume you did this for backwards compatibility, but it leads to issues discussed in #93